### DISCUSSION TITLE
Authorization doesnt work correctly

### DISCUSSION THREAD

**Discussion by MichalNemec** [directus/directus#13415] [Archive]
Hello,
Fresh install of directus with one simple collection (name text).
from app i can log in and get access token.

`GET Http://localhost:8055/items/productCategories?filter%5Bstatus%5D%5B_eq%5D=published&fields=id%2CtextColor%2ChomeWidth%2Cimage%2Ctranslations.languages_code%2Ctranslations.title
Accept: "application/json"
Authorization: Bearer eyJhbGciOiJIUzI1NiIsInR5cCI6IkpXVCJ9.eyJpZCI6IjQzZmFkMTUzLWEzNzYtNDNjYy04MDM2LTUzYWI2Y2RhZDcwYiIsInJvbGUiOiI3Njg1YTdiOS0xY2JkLTQ2MWEtYTJiNS1iMGMyMWEwNGY0ZTMiLCJhcHBfYWNjZXNzIjp0cnVlLCJhZG1pbl9hY2Nlc3MiOnRydWUsImlhdCI6MTY1Mjk0NDM1NiwiZXhwIjoxNjUyOTQ1MjU2LCJpc3MiOiJkaXJlY3R1cyJ9.UDlelPbCVeK5kdxRtjJE6TAiGnFAfK8c6Az68rb6eb4`

leads to
`"errors": [
    {
      "message": "You don't have permission to access this.",
      "extensions": {
        "code": "FORBIDDEN"
      }
    }
  ]`

`GET Http://localhost:8055/items/productCategories?filter%5Bstatus%5D%5B_eq%5D=published&fields=id%2CtextColor%2ChomeWidth%2Cimage%2Ctranslations.languages_code%2Ctranslations.title&access_token=eyJhbGciOiJIUzI1NiIsInR5cCI6IkpXVCJ9.eyJpZCI6IjQzZmFkMTUzLWEzNzYtNDNjYy04MDM2LTUzYWI2Y2RhZDcwYiIsInJvbGUiOiI3Njg1YTdiOS0xY2JkLTQ2MWEtYTJiNS1iMGMyMWEwNGY0ZTMiLCJhcHBfYWNjZXNzIjp0cnVlLCJhZG1pbl9hY2Nlc3MiOnRydWUsImlhdCI6MTY1Mjk0NDM1NiwiZXhwIjoxNjUyOTQ1MjU2LCJpc3MiOiJkaXJlY3R1cyJ9.UDlelPbCVeK5kdxRtjJE6TAiGnFAfK8c6Az68rb6eb4`

leads to 
`"errors": [
    {
      "message": "You don't have permission to access this.",
      "extensions": {
        "code": "FORBIDDEN"
      }
    }
  ]`

when i try it via api-viewer (https://github.com/u12206050/directus-extension-api-viewer-module/releases/tag/1.1.1) module, it works. When i se to public role all settings, it works.

### To Reproduce

1) fresh install
2) create admin user
3) login in postman or whatever to obtain api key
4) Authorization Bearer $token

### Errors Shown

09:20:45 ✨ request completed GET 403 /items/productCategories?filter%5Bstatus%5D%5B_eq%5D=published&fields=id%2CtextColor%2ChomeWidth%2Cimage%2Ctranslations.languages_code%2Ctranslations.title 7ms

09:17:51 ✨ request completed GET 403 /items/productCategories?filter%5Bstatus%5D%5B_eq%5D=published&fields=id%2CtextColor%2ChomeWidth%2Cimage%2Ctranslations.languages_code%2Ctranslations.title&access_token=eyJhbGciOiJIUzI1NiIsInR5cCI6IkpXVCJ9.eyJpZCI6IjQzZmFkMTUzLWEzNzYtNDNjYy04MDM2LTUzYWI2Y2RhZDcwYiIsInJvbGUiOiI3Njg1YTdiOS0xY2JkLTQ2MWEtYTJiNS1iMGMyMWEwNGY0ZTMiLCJhcHBfYWNjZXNzIjp0cnVlLCJhZG1pbl9hY2Nlc3MiOnRydWUsImlhdCI6MTY1Mjk0NDM1NiwiZXhwIjoxNjUyOTQ1MjU2LCJpc3MiOiJkaXJlY3R1cyJ9.UDlelPbCVeK5kdxRtjJE6TAiGnFAfK8c6Az68rb6eb4 4

**Comment by azrikahar** [comment]
Unfortunately I can't seem to reproduce this locally:

https://user-images.githubusercontent.com/42867097/169241390-43ca7bcc-aeff-4227-9b02-28168b2d5531.mp4

Though my URL is much simpler, but the token being used should still stand as it's an admin token. Would you mind verifying/trying the following:

- is the collection actually named as `productCategories` exactly? It is case sensitive when you're requesting it via the API.

- Just to make a differential diagnosis here, can you try making a new collection called `test` (all lowercase) and test `/items/test` instead?

- Can you try making the GET request to `/items/productCategories` without any additional query parameters?

- Can you try using a static token instead (configurable in the admin user's profile page)?

**Reply by MichalNemec** [reply]
i made the test collection and now its working, it would be nice to know then if it cannot find collection to show in directus log, its very confusing, because forbidden forces you to check permissions instead of anything else.

**Reply by azrikahar** [reply]
> i made the test collection and now its working

Thanks for confirming. Does this mean the original `productCategories` was a case sensitivity issue?

> it would be nice to know then if it cannot find collection to show in directus log, its very confusing, because forbidden forces you to check permissions instead of anything else.

Totally understandable. It is something that was brought up in #13083 and #4368 to make it configurable (the current approach was initially set out to err on the side of caution).

EDIT: wrong links!

**Reply by MichalNemec** [reply]
> > i made the test collection and now its working
> 
> Thanks for confirming. Does this mean the original `productCategories` was a case sensitivity issue?
> 
Yes, when i created it (based on my previous tests, it was always productCategories) and now its with capital P, so finally fixed. Took me a while to uncheck all permissions for public :)

Thank you for your help.

Just got another question.. is there somewhere doc for developing directus hooks? because for the life of me i cannot figure out structure of the hook itself (coming from angular when typescript is just amazing). Then we can close this issue.

**Reply by azrikahar** [reply]
Glad it's sorted out! You can refer to the extensions docs to get started: https://docs.directus.io/extensions/creating-extensions/, and you can choose Hooks with the Typescript template 👍 Most operations can be done via the services (you should be able to find them in the examples shown in the docs). 

Admittedlythe actual docs for the services is still not up in the docs yet as there's still an ongoing docs rework in general (currently services docs are tracked in #10523), but you can find their source code over here: https://github.com/directus/directus/tree/main/api/src/services. If you have any doubts/questions in the process, you can also join our [community Discord Server](https://directus.chat) and ask them over there as well. Hope that helps!
